### PR TITLE
tsm: cache: add cache throughput related statistics.

### PR DESCRIPTION
**Pull request by jonseymour**
Complementing and extending the changes in #5758 

Add 2 level statistics:

  * snapshotCount
  * cacheAgeMs

Add 2 counter statistics

  * cachedBytes
  * WALCompactionTimeMs

snapshotCount can be used to measure transient write errors that are causing snapshots to accumulate

cacheAgeMs can be used to guage the level of write activity into the cache

differences between cachedBytes stats sampled at different times can be used to calculate cache throughput rates

(cachedBytes-diskBytes)/WALCompactionTimeMs can be used calculate TSM compaction throughput.

ratio of difference between first and last WAL compaction time over the interval
length is an estimate of percentage of cache throughput consumed.

Signed-off-by: Jon Seymour <jon@wildducktheories.com>

### CODE REVIEW

**Comment by jonseymour** [comment]
G'day @mark-rushakoff  - this pull request against your mr-disk-stats branch extends your work to add some additional cache statistics that I think would be useful. See discussion in #5499  for more details.

(I submitted the pull request for this against your branch, because my work is necessarily dependent upon your work - feel free to rebase and squash onto the tip of your branch, if you so desire.)

/cc @jwilder

File: tsdb/engine/tsm1/cache.go [hunk 1]
```diff
@@ -67,8 +68,17 @@ func (e *entry) deduplicate() {
...
+
+	statCacheMemoryBytes = "memBytes"      // level: Size of in-memory cache in bytes
+	statCacheDiskBytes   = "diskBytes"     // level: Size of on-disk snapshots in bytes
+	statSnapshots        = "snapshotCount" // level: Number of active snapshots.
+	statCacheAge         = "cacheAge"      // level: milliseconds since cache was last snapshotted at sample time
+
+	// counters - accumulative measures
+
+	statCachedBytes       = "cachedBytes"       // counter: Total number of bytes written into snapshots.
+	statWALCompactionTime = "WALCompactionTime" // counter: total number of milliseconds spent compacting snapshots
```
Comment (mark-rushakoff) [inline tsdb/engine/tsm1/cache.go]: Would you mind adding the units to the stat names? i.e. `WALCompactionTimeMs` and `cacheAgeMs`.

The comments you've added (and your PR notes as well) are very clear and informative, but having the unit in the name of the stat eliminates some potential guesswork.

**Comment by mark-rushakoff** [comment]
@jonseymour One small naming change requested, then I'm happy to accept this PR. I'll work with the database team next week to get this merged.

Thanks for the contribution and the detailed notes.

**Comment by jonseymour** [comment]
Sure will update when I am next at keyboard.

**Comment by jonseymour** [comment]
@mark-rushakoff done.

**Pull request merged**